pwiz_tools/Skyline/FileUI/PeptideSearch/TransitionSettingsControl.cs

```diff
@@ -349,11 +352,30 @@ public TransitionSettings GetTransitionSettings(Form parent)
                 settings.Instrument.MaxInclusions, settings.Instrument.MinTime, settings.Instrument.MaxTime);
             Helpers.AssignIfEquals(ref instrument, settings.Instrument);
 
-            TransitionLibraryPick pick = settings.Libraries.Pick != TransitionLibraryPick.none ? settings.Libraries.Pick : TransitionLibraryPick.all;
+            var pick = settings.Libraries.Pick != TransitionLibraryPick.none && DocumentSettings.PeptideSettings.Libraries.LibrarySpecs.Any(lib => !lib.IsDocumentLibrary)
+                ? settings.Libraries.Pick
+                : TransitionLibraryPick.filter;
+            if (CountFinder != null && pick == TransitionLibraryPick.filter)
+                pick = TransitionLibraryPick.all;
             var libraries = new TransitionLibraries(ionMatchTolerance, minIonCount, ionCount, pick);
             Helpers.AssignIfEquals(ref libraries, settings.Libraries);
 
             return new TransitionSettings(settings.Prediction, filter, libraries, settings.Integration, instrument, settings.FullScan, settings.IonMobilityFiltering);
         }
+
+        private IEndCountFragmentFinder CountFinder => TransitionFilter.GetEndFragmentFinder(
+            TransitionFilter.GetEndFragmentNameFromLabel(IonFilter ? IonRangeTo : DocumentTransitionSettings.Filter.FragmentRangeLast.Label)
+        ) as IEndCountFragmentFinder;
+
+        private void comboRangeTo_SelectedIndexChanged(object sender, System.EventArgs e)
+        {
+            var countFinder = CountFinder;
+            if (countFinder != null)
+            {
+                IonCount = countFinder.Count;

```

Comment:
This is a tricky bit of code. Even though, it may all work out, I think you should not use IonCount as much as you do, since in each reference it is doing int.Parse() of the text value, which you set at the top. So, it is unlikely to throw a FormatException. But, you probably should assign to an integer local variable at the top, and then use that below.

int ionCount = countFinder.Count;
IonCount = ionCount;